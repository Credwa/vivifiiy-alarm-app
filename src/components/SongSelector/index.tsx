import React from 'react';
import ScrollSelector from '@/components/ScrollSelector';
import Colors from '@/constants/Colors';
import VivText from '@/components/VivText';
import { smallScreenWidthBreakpoint, largeScreenWidthBreakpoint } from '@/constants/Values';
import { Dimensions } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

const windowWidth = Dimensions.get('window').width;

interface SongSelectorProps {
  data: any[];
}

export default function SongSelector({ data }: SongSelectorProps) {
  return (
    <>
      <VivText
        fontName={
          windowWidth < smallScreenWidthBreakpoint
            ? 'Body'
            : windowWidth > largeScreenWidthBreakpoint
            ? 'Title3'
            : 'Title5'
        }
        color={Colors.greyLight1}
      >
        Up next...
      </VivText>
      <ScrollSelector
        dataSource={data}
        overrideFontName={
          windowWidth < smallScreenWidthBreakpoint
            ? ['Subhead', 'Caption']
            : windowWidth > largeScreenWidthBreakpoint
            ? ['Title1', 'Title2']
            : ['Headline', 'Footnote']
        }
        selectedIndex={1}
        onValueChange={(data) => {}}
        wrapperWidth={windowWidth}
        wrapperHeight={wp('50%')}
        itemHeight={
          windowWidth < smallScreenWidthBreakpoint ? 45 : windowWidth > largeScreenWidthBreakpoint ? 110 : 50
        }
        highlightColor={Colors.white}
        highlightBorderWidth={0.001}
      />
    </>
  );
}
