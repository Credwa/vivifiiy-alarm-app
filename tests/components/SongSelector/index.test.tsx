import React from 'react';
import renderer from 'react-test-renderer';
import SongSelector from '@/components/SongSelector';

describe('<SongSelector />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<SongSelector data={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
