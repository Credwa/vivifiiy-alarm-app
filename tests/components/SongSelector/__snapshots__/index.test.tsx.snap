// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongSelector /> renders correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#869DB2",
        },
        Object {
          "fontFamily": "Nunito_400Regular",
          "fontSize": 22,
        },
        Object {
          "color": "#CEE1F2",
        },
        undefined,
      ]
    }
  >
    Up next...
  </Text>,
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": null,
        "height": 375,
        "overflow": "hidden",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#ffffff",
          "borderBottomWidth": 0.001,
          "borderTopColor": "#ffffff",
          "borderTopWidth": 0.001,
          "height": 50,
          "position": "absolute",
          "top": 162.5,
          "width": "100%",
        }
      }
    />
    <RCTScrollView
      bounces={false}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      scrollEventThrottle={400}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "height": 162.5,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "height": 162.5,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>,
]
`;
